dot/state/block.go

```diff
@@ -350,11 +377,53 @@ func (bs *BlockState) BestBlockHeader() (*types.Header, error) {
 	return bs.GetHeader(bs.BestBlockHash())
 }
 
+// BestBlockNumber returns the block number of the current head of the chain
+func (bs *BlockState) BestBlockNumber() (*big.Int, error) {
+	header, err := bs.GetHeader(bs.BestBlockHash())
+	if err != nil {
+		return nil, err
+	}
+
+	return header.Number, nil
+}
+
 // BestBlock returns the current head of the chain
 func (bs *BlockState) BestBlock() (*types.Block, error) {
 	return bs.GetBlockByHash(bs.BestBlockHash())
 }
 
+// GetSlotForBlock returns the slot for a block
+func (bs *BlockState) GetSlotForBlock(hash common.Hash) (uint64, error) {
```

Comment:
can we reduce the number of return statements in `GetSlotForBlock`? Maybe extract decode logic?

Reply:
will take a look!

Reply:
I think it's okay as it is, there aren't any other functions in the state that would be using that functionality so I'm not sure it makes sense to break it out.